rules/table-of-contents.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+const visit = require('unist-util-visit');
+const chalk = require('chalk');
+
+module.exports = (ast, file, language, done) => {
+	visit(ast, 'heading', (node, index) => {
+		const title = node.children[0].value;
+
+		if (index === 1 && title !== 'Contents') {
```

Comment:
This algorithm will be more solid if you first find all `heading` nodes:

```js
var headings = ast.children.filter(n => n.type === 'heading');
```

Now, if there’s a markdown heading as the first heading, ignore it for this algorithm:

```js
if (headings[0] && headings[0].depth === 1) { headings.shift() }
```

(thus supporting both `<h1>Title` and `# Title`);

Then, check that `headings[0]` has a `contents` value:

```js
var value = headings[0] && toString(headings[0]);

if (value !==  'Contents') { /* warn */ }
```

Where `toString` comes from [`mdast-util-to-string`](https://github.com/wooorm/mdast-util-to-string). The code you’re currently using can namely throw if an empty heading is found (`## `), or if there’s a `strong` or something else in the heading (`## **Contents**`), and `mdast-util-to-string` prevents that!